drivers/rtc/rtc_atmel_sam.c

```diff
@@ -0,0 +1,651 @@
+/*
+ * RTC driver for the ATMEL SAM family
+ *
+ * Copyright 2022 Bjarki Arge Andreasen
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/*************************************************************************************************/
+/*                                          Dependencies                                         */
+/*************************************************************************************************/
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <zephyr/irq.h>
+#include <zephyr/drivers/rtc.h>
+
+#include <time.h>
+#include <string.h>
+#include <soc.h>
+
+/*************************************************************************************************/
+/*                                             Macros                                            */
+/*************************************************************************************************/
+#define TO_BCD(value) (((((uint32_t)(value)) / 10) << 4) + ((value) % 10))
+
+#define FROM_BCD(value) ((((value) >> 4) * 10) + ((value) & BIT_MASK(4)))
+
+#define RTC_REG_GET_FIELD(value, field) ((RTC_##field##_Msk & value) >> RTC_##field##_Pos)
+
+#define RTC_REG_GET_FIELD_BCD(value, field) FROM_BCD(RTC_REG_GET_FIELD(value, field))
+
+/*************************************************************************************************/
+/*                                           Definitions                                         */
+/*************************************************************************************************/
+#define DT_DRV_COMPAT                           atmel_sam_rtc
+
+/*************************************************************************************************/
+/*                                             Types                                             */
+/*************************************************************************************************/
+typedef void (*rtc_atmel_sam_irq_init_fn_ptr)(void);
+
+struct rtc_atmel_sam_config {
+	uint16_t irq_num;
+	rtc_atmel_sam_irq_init_fn_ptr irq_init_fn_ptr;
+};
```

Comment:
You should use structure `Rtc` to access fields. You can look at https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/watchdog/wdt_sam.c for reference. This means that you must remove all direct register access.

Reply:
Updated to use Rtc struct like adc_sam_afec